File: InternHub/InternHub.Model/Company.cs
using InternHub.Model.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InternHub.Model
{
    public class Company : User, ICompany
    {
        public Guid StateId { get; set; }
        public State State { get; set; }
        public string Name { get; set; }
        public string Website { get; set; }
        public bool IsAccepted { get; set; }
    }
}

File: internhub-frontend/src/models/Company.js
export class Company {
  constructor({ id = "", website = "", address = "", name = "" }) {
    this.id = id;
    this.website = website;
    this.address = address;
    this.name = name;
  }

  static fromJson(json) {
    return new Company({
      id: json["Id"],
      website: json["Website"],
      address: json["Address"],
      name: json["Name"],
    });
  }
}
